.editorconfig

```diff
@@ -0,0 +1,16 @@
+# top level config
+root = true
+
+# All files:
+# utf-8
+# Unix-style newlines
+# Remove end-of-file newlines
+# Soft indents (2 spaces)
+# Don't trim trailing whitespace
```

Comment:
@Alys I can't find the code I was thinking of when I wrote that, but I suspect I meant something like this:

```pug
| Here is some text with a
a(href="#") link
| in it
```

which renders as `Here is some text with alinkin it`, and needs a space at the end of the first line and at the start of the third line to render properly. However, the proper way to do the above is

```pug
| Here is some text with a #[a(href="#") link] in it
```

and Habit barely uses inline text anyway, so maybe the special case for Jade/Pug files shouldn't be included. If it turns out that having trailing whitespace is useful after all, we can always change the editorconfig.